### DISCUSSION TITLE
Kitty background color changed?

### DISCUSSION THREAD

**Discussion by xfzv** [mikesmithgh/kitty-scrollback.nvim#96] [Q&A] [answered]
![01](https://github.com/mikesmithgh/kitty-scrollback.nvim/assets/78810647/fc0c4aba-7e8a-484b-a64c-c52631344212)

My Kitty color background is `#1e1e20`. I've just noticed that when triggering `kitty-scrollback` with either `kitty_mod+h` or `kitty_mod+g`, it changes to `#1c1d23`.

I'm pretty sure it was sticking to the Kitty background color until recently, but maybe I'm wrong.

I tried to downgrade to older versions (as far as v2.0.0), same behavior. Also occurs without loading my Kitty theme config file:

![02](https://github.com/mikesmithgh/kitty-scrollback.nvim/assets/78810647/0ddc8ad6-4cc0-46e7-9702-80aea4a597de)

Maybe it's forced somehow and it's completely normal? 

My config:

- `kitty-scrollback.lua`
```lua
return {
  "mikesmithgh/kitty-scrollback.nvim",
  cmd = { "KittyScrollbackGenerateKittens", "KittyScrollbackCheckHealth" },
  event = "User KittyScrollbackLaunch",
  opts = {},
}
```

- `kitty-scrollback-config.lua`
```lua
vim.opt.runtimepath:append(vim.fn.stdpath("data") .. "/lazy/kitty-scrollback.nvim")

require("kitty-scrollback").setup({
  global = function()
    return {
      visual_selection_highlight_mode = "reverse", -- most likely not related because it only affects selection, but I tried without this, or with "darken", "kitty", "nvim" and got the same behavior
      status_window = {
        icons = {
          nvim = "",
        },
      },
    }
  end,
})
```

```sh
$ kitty --version
kitty 0.31.0 created by Kovid Goyal
```

- `kitty.conf`
```
# kitty-scrollback.nvim Kitten alias
action_alias kitty_scrollback_nvim kitten /home/xfzv/.local/share/nvim/lazy/kitty-scrollback.nvim/python/kitty_scrollback_nvim.py --nvim-args -u /home/xfzv/.config/nvim/lua/kitty-scrollback-config.lua

# Browse scrollback buffer in nvim
map kitty_mod+h kitty_scrollback_nvim

# Browse output of the last shell command in nvim
map kitty_mod+g kitty_scrollback_nvim --config ksb_builtin_last_cmd_output

# Show clicked command output in nvim
mouse_map ctrl+shift+right press ungrabbed combine : mouse_select_command_output : kitty_scrollback_nvim --config ksb_builtin_last_visited_cmd_output


include themes/catppuccin-mocha.conf
```
- `~/.config/kitty/themes/catppuccin-mocha.conf`
```
foreground                    #CDD6F4
background                    #1E1E20
selection_foreground          none
selection_background          none

cursor                        none
url_color                     #89B4FA

active_tab_foreground         #CDD6F4
active_tab_background         #505052
inactive_tab_foreground       #CDD6F4
inactive_tab_background       #262626

active_border_color           #A6E3A1
inactive_border_color         #A6ADC8

# black
color0                        #45475A
color8                        #585B70

# red
color1                        #F38BA8
color9                        #F38BA8

# green
color2                        #A6E3A1
color10                       #A6E3A1

# yellow
color3                        #F9E2AF
color11                       #F9E2AF

# blue
color4                        #89B4FA
color12                       #89B4FA

# magenta
color5                        #B4BEFE
color13                       #B4BEFE

# cyan
color6                        #94E2D5
color14                       #94E2D5

# white
color7                        #BAC2DE
color15                       #A6ADC8
```

**Comment by xfzv** [comment]
I've just tried in a virtual machine and the background color doesn't change.

Might be a Neovim change upstream, I'll investigate and report later.

---

Edit: Indeed, it looks like it's due to a recent Neovim change.

On the VM with Neovim 0.9.4, the background color doesn't change. I tried with 0.10 which I'm also running on my main system and got the same behavior.

There have been a few commits related to colors in the last few days, maybe related?

https://github.com/neovim/neovim/commit/64a14026d76ba1798d91e15a941fcb6af7cbc5ad
https://github.com/neovim/neovim/commit/2613ba5000d4c0d9b15e2eec2d2b97615575925e
https://github.com/neovim/neovim/commit/8f10362cdc5835a1cd86c8195ce8e3b10ab85384
https://github.com/neovim/neovim/commit/a5a346678a8211ea07f318de42e557ad3909f65e
https://github.com/neovim/neovim/commit/6c3ddd8c510a28d4acb777b9676a0d4137b309fa

**Comment by mikesmithgh** [comment, marked as answer]
Hey @xfzv thanks for the detailed information! Yes, it is due to the new default neovim coloscheme. See https://github.com/neovim/neovim/pull/26334#issuecomment-1835795626

I have a ticket to fix this https://github.com/mikesmithgh/kitty-scrollback.nvim/issues/93

There are a couple ideas there you could play with. 

You might also want to try `:colorscheme vim` that is the previous default theme.

**Comment by mikesmithgh** [comment]
Awesome screenshots and details 😸 👍

**Comment by xfzv** [comment]
I checked issues before opening this discussion but I thought that #93 was something else.

Sorry! :bow:
